mdp/test/test_nodes_generic.py

```diff
@@ -192,6 +192,8 @@ def test_dtype_consistency(klass, init_args, inp_arg_gen,
     supported_types = klass(*args).get_supported_dtypes()
     for dtype in supported_types:
         inp = inp_arg_gen()
+        if klass.__name__ == "ComplementNBScikitsLearnNode":
+            inp = numx.absolute(inp)
```

Comment:
Out of curiosity: Is there a known rationale why ComplementNBScikitsLearnNode stopped supporting negative input? Or did it never support it but just did not check properly?

Reply:
My understanding is, that the code for these tests was [implemented prior to 2017](https://github.com/mdp-toolkit/mdp-toolkit/commits/master/mdp/test/test_nodes_generic.py).
It seems like the nodes implemented are automatically wrapping multiple scikit-learn classes. The [Complement Naive Bayes method joined these classes Aug 28, 2017](https://github.com/scikit-learn/scikit-learn/commit/a571b01ac8a92504c522a860993428c0f46d7b29#diff-dff46bc51c084c731bcbe4cb9b6aa243) and appeared in [scikits-learn 0.20.0 for the first time](https://scikit-learn.org/stable/whats_new.html#sklearn-naive-bayes) .

Thus, this node was likely included automatically and requires more specific input than it seems to have been planned for. Ideally, there should be an issue for this or a todo-list entry for coming up with a nice fix that lasts.

Reply:
Okay, thanks for the clarification. Would something speak against adding some [excludes](https://github.com/mdp-toolkit/mdp-toolkit/blob/master/.travis.yml#L23) for Python 2.7 and newer scikits? At least for a while?

Reply:
I see. From my view, a more generic approach would be to have a set of node names that require positive input. Call numx.absolute if the present node is in that set. Given that there is only one node that needs this special treatment, I agree such an approach would be overkill. So for now I think your approach is the best way to resolve it. I would just like to see a comment somwhere in the code close to the special casing, saying that if more nodes show up with this demand, we will switch to a more generic solution. The insights from [your comment above](https://github.com/mdp-toolkit/mdp-toolkit/pull/47#discussion_r295247636) should also go into that comment. Maybe a link to this discussion as well.